### PR TITLE
fix(compiler): fix validation for fragment spread of interface without implementers

### PR DESCRIPTION
**Pull request by goto-bus-stop**
This `Intf` has no implementations. As written in the spec, doing a `... on Intf` fragment spread should never work, as the set of possible types is empty and can never intersect with the parent type. However, implementations like graphql-js and graphql-go have an early check, accepting the fragment if the type condition is equal to the parent type. This tests reproduces that.

We may want to align with graphql-js and graphql-go rather than the spec here for compatibility? Though it's not something that's likely to happen in the real world.

Ref https://github.com/graphql/graphql-spec/issues/1109

### CODE REVIEW

File: crates/apollo-compiler/test_data/diagnostics/0091_recursive_interface_definition.graphql [hunk 1]
```diff
@@ -4,13 +4,17 @@ interface A implements B {
 interface B implements A {
   name: String
 }
-fragment recursive on A {
-  name
+type Impl implements A & B {
+  name: String
 }
```
Comment (goto-bus-stop) [inline crates/apollo-compiler/test_data/diagnostics/0091_recursive_interface_definition.graphql:9]: Unrelated change, just fixing the schema so it doesn't rely on having an interface with 0 implementers

**Comment by duckki** [comment]
I agree that Apollo's validator is spec-compliant currently, while others are not :)

At the same time, this could be an oversight in the spec. It's good to be able to test queries before writing an implementation (despite the weirdness of conditioning on the parent type).

**Review by lrlna — APPROVED** [review]
